### DISCUSSION TITLE
Issue with [@types/emscripten]

### DISCUSSION THREAD

**Discussion by damms005** [DefinitelyTyped/DefinitelyTyped#68690] [Issues with a @types package]
I got compilation errors in my TypeScript project:

```
npm run compile                           

> devdb@1.4.0 compile
> tsc -p ./

node_modules/@types/emscripten/index.d.ts:44:33 - error TS2304: Cannot find name 'WebGLRenderingContext'.

44     preinitializedWebGLContext: WebGLRenderingContext;
                                   ~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/emscripten/index.d.ts:58:28 - error TS2304: Cannot find name 'MessageEvent'.

58     onCustomMessage(event: MessageEvent): void;
                              ~~~~~~~~~~~~


Found 2 errors in the same file, starting at: node_modules/@types/emscripten/index.d.ts:44
```

## How to reproduce
- `git clone https://github.com/damms005/devdb-vscode`
- `git checkout chore/switch-sqlite3-to-sqljs`
- `npm i`
- `npm run compile`
- See the error above

How do I get the missing types?

[a comment by an automated account is not shown]

**Comment by aleksander-ciesielski** [comment]
just encountered the same issue. I believe both `WebGLRenderingContext` and `MessageEvent` are part of Web APIs, so I included DOM libs in my project (by adding `"dom"` to `compilerOptions.lib` in the `tsconfig.json`), which fixed the issue for me:
```json
{
  "compilerOptions" {
    "lib": ["dom"],
  }
}
```
